plugins/coverbrowser.koplugin/main.lua

```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
     end
     sub_item_table[#modes].separator = true
     table.insert(sub_item_table, {
-        text = _("Use this mode everywhere"),
+        text = _("Hide book covers"),
+        enabled_func = function()
+            return not fc.display_mode_type == false
+        end,
         checked_func = function()
-            return BookInfoManager:getSetting("unified_display_mode")
+            return filemanager_display_mode == "mosaic_text" or filemanager_display_mode == "list_only_meta"
         end,
         callback = function()
-            if BookInfoManager:toggleSetting("unified_display_mode") then
-                self:setupHistoryDisplayMode(filemanager_display_mode)
-                self:setupCollectionDisplayMode(filemanager_display_mode)
-            end
+            if filemanager_display_mode == "mosaic_image" then
+                self:setupFileManagerDisplayMode("mosaic_text")
+            elseif filemanager_display_mode == "mosaic_text" then
+                self:setupFileManagerDisplayMode("mosaic_image")
+            elseif filemanager_display_mode == "list_only_meta" then
+                self:setupFileManagerDisplayMode("list_image_meta")
+            else
+                self:setupFileManagerDisplayMode("list_only_meta")
+            end  
         end,
     })
     table.insert(sub_item_table, {
-        text = _("History display mode"),
+        text = _("Replace book title with file name"),
         enabled_func = function()
-            return not BookInfoManager:getSetting("unified_display_mode")
+            return fc.display_mode_type == "list"
         end,
-        sub_item_table = history_sub_item_table,
-    })
-    table.insert(sub_item_table, {
-        text = _("Collections display mode"),
-        enabled_func = function()
-            return not BookInfoManager:getSetting("unified_display_mode")
+        checked_func = function()
+            return filemanager_display_mode == "list_image_filename"
+        end,
+        callback = function()
+            if filemanager_display_mode == "list_image_filename" then
+                self:setupFileManagerDisplayMode("list_image_meta")
+            else
+                self:setupFileManagerDisplayMode("list_image_filename")
+            end
         end,
-        sub_item_table = collection_sub_item_table,
+        separator = true,
     })
+    ------------------------------------------------------------
+    -- add these settings to File browser Settings submenu
+    if menu_items.filebrowser_settings == nil then return end
+    table.insert(menu_items.filebrowser_settings.sub_item_table, 9, {
```

Comment:
It's the last item, insertion index not needed.

Reply:
good point.